packages/compiler/src/core/types.ts

```diff
@@ -1897,9 +1898,16 @@ export interface LibraryInstance {
   entrypoint: JsSourceFileNode;
   metadata: LibraryMetadata;
   definition?: TypeSpecLibrary<any>;
+}
+
+export interface LinterLibraryInstance extends LibraryInstance {
   linter: LinterResolvedDefinition;
 }
 
+export interface TransformerLibraryInstance extends LibraryInstance {
+  transformer: TransformerResolvedDefinition;
+}
```

Comment:
What's the reason for making these separate interfaces (instead of adding `transformer` to `LibraryInstance`)?

I only see them being used in intersection (except for tests, but I don't think that should dictate the type system).